R/construction.R

```diff
@@ -1,66 +1,74 @@
 arrow_to_rdf = function(at, schema = NULL, schema_overrides = NULL, rechunk = TRUE) {
   # new column names by schema, #todo get names if schema not NULL
-  original_schema = schema
+  n_cols = at$num_columns
+
   new_schema = unpack_schema(
     schema = schema %||% at$ColumnNames(),
     schema_overrides = schema_overrides
   )
   col_names = names(new_schema)
 
-  if (length(col_names) != at$num_columns) stop("schema length does not match arrow table")
-
+  if (length(col_names) != n_cols) stop("schema length does not match arrow table")
 
-  # store special translated columns here
-  special_cols = new.env(parent = emptyenv())
+  data_cols = list()
+  # dictionaries cannot be built in different batches (categorical does not allow
+  # that) so we rechunk them and create them separately.
+  # struct columns don't work properly if they contain multiple chunks.
+  special_cols = list()
 
   ## iter over columns, possibly do special conversion
-  i_col = 0
-  for (column in at$columns) {
-    i_col = i_col + 1
-    name = col_names[i_col]
+  for (i in 1L:n_cols) {
```

Comment:
can you use `seq_along(n_cols)` just for safety?

Reply:
> can you use `seq_along(n_cols)` just for safety?

n_cols has length 1, so wouldn't `seq_along(n_cols)` always return 1?

Reply:
Right it should be `seq_len()` instead